app/actors/hyku_addons/actors/note_field_actor.rb

```diff
@@ -15,14 +15,15 @@ def update(env)
       private
 
         def serialize_note_field(env)
+          return unless env.curation_concern.respond_to?(:note)
+
           if env.attributes[:note].present?
```

Comment:
I am assuming here that `env.curation_concern.note` is an array, as it has another array appended in this code.  Given that a more terse way of writing this might be:
````
def serialize_note_field(env)
  return unless env.curation_concern.respond_to?(:note)
  
  @email = env.user.email if env.attributes[:note].present?
    
  env.attributes[:note] = note_array(env).push(*process_note_hash(env))
end

def note_array(env)
  env.curation_concern.note || []
end

def process_note_hash(env)
  return unless env.attributes[:note]

  Array(
    { email: email, timestamp: Time.zone.now.to_s, note: env.attributes[:note] }.to_json
  )
end
````

This is untested though and may contain typos or other errors!  It's also possible to refactor this down more if `@email` is only being set here for use in `process_note_hash`.  NB This will return `[]` not `nil` if `env.curation_concern.note.nil?` and `env.attributes[:note].nil?` which might not be what you want.

Reply:
Thanks for this @thenapking. The only line I actually needed was here https://github.com/ubiquitypress/hyku_addons/pull/512/files/97c0d1ef8ee8f06c210594dd93d248445ea12ed7#diff-f4cd9cb2f565f29a5825dbcd96ca79b2082a0871a4f39cef36f4d8bea91699d6R18

Which was just to ensure that the save process wouldn't fail when a note field was missed. The rest is just styling the variable assignment to be indented and easier to read. 

If you want to put in a PR with the changes above then that would be great :)